src/core/Delta.ts

```diff
@@ -31,32 +31,103 @@ export enum DeltaQueue {
   kUnknown
 }
 
+/**
+ * Output JSON format for delta.json and deltaLog.json based on feedback
+ * from the AWG on 8/22/2023 to keep the output simple
+ * 
+ * So internally, we are storing the full CveCorePlus, but externally,
+ * and specifically when writing out to JSON, we are using this shortened format
+ 
+ * see https://github.com/CVEProject/cvelistV5/issues/23 for some additional discussions
+ * before and after the AWG meeting on 8/22
+ */
+export class DeltaOutpuItem {
+
+  static _cveOrgPrefix = `https://www.cve.org/CVERecord?id=`;
+  static _githubRawJsonPrefix = `https://raw.githubusercontent.com/CVEProject/cvelistV5/main/cves/`
+
+  cveId: string;          // string version of the CVE ID
+  cveOrgLink?: string;    // url to cve.org record
+  githubLink?: string;    // url to Github raw json
+  dateUpdated?: string;   // ISO string
+
+  static fromCveCorePlus(cvep: CveCorePlus): DeltaOutpuItem {
+    let deltaItem = new DeltaOutpuItem();
+    const cveid = cvep.cveId.toString();
+    deltaItem.cveId = cveid;
+    deltaItem.cveOrgLink = `${DeltaOutpuItem._cveOrgPrefix}${cveid}`;
+    deltaItem.githubLink = `${DeltaOutpuItem._githubRawJsonPrefix}${CveId.getCveDir(cveid)}/${cveid}.json`;
+    deltaItem.dateUpdated = cvep.dateUpdated;
+    return deltaItem;
+  }
+
+  static replacer(key: string, value: any) {
+    let items = [];
+    if (key === 'new' || key === 'updated' || key === 'unknown') {
+      value.forEach((item: CveCorePlus) => {
+        // Note, it is important to keep this loop as simple as possible,
+        //  don't rely on item being an actual CveCorePlus object since
+        //  it may not be depending on what built it
+        // console.log(`replacer item=${JSON.stringify(item, null, 2)}`);
+        const cveid = item.cveId.toString();
+        if (cveid) {
+          items.push({
+            cveId: cveid,
+            cveOrgLink: `${DeltaOutpuItem._cveOrgPrefix}${cveid}`,
+            githubLink: `${DeltaOutpuItem._githubRawJsonPrefix}${CveId.getCveDir(cveid)}/${cveid}.json`,
+            dateUpdated: item.dateUpdated
+          });
+        }
+      });
+      return items;
+    }
+    else {
+      return value;
+    }
+  }
+
+  toJSON() {
+    return {
+      cveId: this.cveId,
+      cveOrgLink: this.cveOrgLink,
+      githubLink: this.githubLink,
+      dateUpdated: this.dateUpdated
+    };
+  }
+
+
+}
+
 
-export class Delta /*implements DeltaProps*/ {
+export class Delta {
+
+  fetchTime?: string;
+  durationInMsecs?: number;   // if not set, it means that it was not calculated
   numberOfChanges: number = 0;
-  // published: CveCore[] = [];
-  new: CveCore[] = [];
-  updated: CveCore[] = [];
-  unknown: CveCore[] = [];
+  new: CveCorePlus[] = [];
+  updated: CveCorePlus[] = [];
+  unknown?: CveCorePlus[] = []; // for any CVE that is not new or updated, which should never be the case except for errors
```

Comment:
What exactly does that mean? When would a CVE be included in the unknown category of updates?

Reply:
Good question.  The history is from when this file was for debugging only, where it served as a catch-all for anything not "new" or "updated".  It hasn't been populated since the bulk download launch in March, and I thought of taking it out, but chose to keep it in as a way to catch any errors.

Reply:
Right, but what errors would cause a CVE to appear in this entry? If the category is there, then it should be clear to consumers of the file on how to handle any entries in it. If I process a log entry and see something in `unknown`, what should I do?

Reply:
Good point.  The algorithm currently puts PUBLISHED and REJECTED CVEs into the "new" list, and UPDATED CVEs into "updated" list.  This is because the algorithm uses git to evaluate the categories based on whether the CVE json file is A (added), M (modified), and since both published and rejected are both added files from git's point of view, they are placed into the "new" list.  There are other git statuses:  e.g., deleted, moved, etc.  Since these can't happen in the repository, if any of these are found, they would be placed into "unknown".

Your point is very well taken, however.  The list should reflect how the file is to be used from a CVE point of view, instead of a git/implementation point of view.  What do you think if I categorize the list into the following, I think it would represent a more CVE-oriented (and useful) way of categorizing:

```typescript
published: CveCorePlus[]; // newly added CVE files with state of PUBLISHED
updated: CveCorePlus[];   // modified CVE files with state of UPDATED (these were previously PUBLISHED or UPDATED files)
rejected: CveCorePlus[];  // newly added CVE files with state of REJECTED
error: CveCorePlus[];     // any CVE files that do not fall into the 3 categories above
                          //   from git's point of view (e.g., renamed, deleted)
                          //   or from CVE's point of view (e.g., RESERVED)
```

Would this make more sense and be more useful?

Reply:
I think it's fine to keep it as new and updated, splitting it out to more categories just means I have to concatenate more lists :-) My comment was mostly around the expectation of the unknown list. As a consumer, if I have a local DB of CVE records and I want to keep them updated, I'll simply process the CVE record from either of these lists as "needing an update" and whether it ends up being a create or update operation is irrelevant and up to the underlying DB query.

So to keep things simple, I would propose just two categories: new and updated. If a file is created for the first time in the repo, add it to "new", anything else goes to "updated".

One more question: can a CVE record ever actually be removed from the data set (file deleted)? Same goes for renames, when might that happen?

Reply:
OK, I'll keep it as "new" and "updated", and just rename "unknown" to "error" as a catch all so that it is more specific about its purpose.

From what I understand from various conversations/meetings, no CVE JSON file should ever be deleted or renamed in the cvelistV5 repository.  There is no function in the cvelist-bulk-download code that would do this currently, and none expected to either.